File: backend/src/main/java/com/procredito/backend/service/DashboardService.java
package com.procredito.backend.service;

import com.procredito.backend.dto.AnalistaResumenResponse;
import com.procredito.backend.dto.DashboardResponse;

import java.util.List;

public interface DashboardService {

    /**
     * Resumen del dashboard según el rol del usuario autenticado:
     * - ADMIN: totales de todos.
     * - ANALISTA: solo sus totales.
     */
    DashboardResponse obtenerResumen();

    /**
     * Resumen por analista (solo ADMIN).
     * Devuelve una fila por cada analista con sus métricas individuales.
     */
    List<AnalistaResumenResponse> obtenerResumenPorAnalista();
}

File: backend/backend/frontend/src/app/core/services/dashboard.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { DashboardResumen, AnalistaResumen } from '../models';

@Injectable({ providedIn: 'root' })
export class DashboardService {

  private readonly http = inject(HttpClient);
  private readonly apiUrl = `${environment.apiUrl}/dashboard`;

  /**
   * Resumen general. Según rol:
   * - ADMIN: totales globales.
   * - ANALISTA: solo sus totales.
   */
  obtenerResumen(): Observable<DashboardResumen> {
    return this.http.get<DashboardResumen>(`${this.apiUrl}/resumen`);
  }

  /**
   * Resumen por analista. Solo ADMIN.
   */
  obtenerResumenPorAnalista(): Observable<AnalistaResumen[]> {
    return this.http.get<AnalistaResumen[]>(`${this.apiUrl}/por-analista`);
  }
}
